install/install.ps1

```diff
@@ -8,87 +8,183 @@ $script:ErrorActionPreference='Stop'
+    $client = New-Object System.Net.WebClient
+    $client.DownloadFile($releaseUrl, $zipFile)
+
+    # A running instance of azureauth can cause installation to fail, so we try to kill any running instances first.
+    # We suppress taskkill output here because this is a best effort attempt and we don't want the user to see its output.
+    # Here, Get-Process is used to first determine whether there is an existing azureauth process. If there is, kill the existing process first.
+    $ProcessCheck = Get-Process -Name azureauth -ErrorAction SilentlyContinue -ErrorVariable ProcessError
+    if ($ProcessCheck -ne $null)
+    {
+        Write-Verbose "Stopping any currently running azureauth instances"
+        taskkill /f /im azureauth.exe 2>&1 | Out-Null
+
+        # After killing the process it is still possible for there there to be locks on the files it was using (including
+        # its own DLLs). The OS may take an indeterminate amount of time to clean those up, but so far we've observed 1
+        # second to be enough.
+        Start-Sleep -Seconds 1
     }
-    Set-ItemProperty -Path $registryPath -Name PATH -Value $newPath
+
+    if (Test-Path -Path $extractedDirectory) {
+        Write-Verbose "Removing pre-existing extracted directory at ${extractedDirectory}"
+        Remove-Item -Force -Recurse $extractedDirectory
+    }
+
+    Write-Verbose "Extracting ${zipFile} to ${extractedDirectory}"
+    Add-Type -AssemblyName System.IO.Compression.FileSystem
+    [System.IO.Compression.ZipFile]::ExtractToDirectory($zipFile, $azureauthDirectory)
+
+    Write-Verbose "Removing ${zipFile}"
+    Remove-Item -Force $zipFile
+
+    $registryPath = 'Registry::HKEY_CURRENT_USER\Environment'
+    $currentPath = (Get-ItemProperty -Path $registryPath -Name PATH -ErrorAction SilentlyContinue).Path
+
+    # Permanently add the latest directory to the current user's $PATH (if it's not already there).
+    # Note that this will only take effect when a new terminal is started.
+    if (($currentPath -ne $null) -And !$currentPath.Contains($azureauthDirectory)) {
+        Write-Verbose "Updating `$PATH to include ${versionPath}"
+        $newPath = if ($currentPath -Match $null) {
+            "${versionPath}"
+        } else {
+            "${currentPath};${versionPath}"
+        }
+        Set-ItemProperty -Path $registryPath -Name PATH -Value $newPath
+    }
+    else{
+        $paths = $currentPath.Split(";")
+        $newPath = "";
+        $azureauthSource = (get-command azureauth -ErrorAction SilentlyContinue).Source
+        ForEach($path in $paths){
+            if(!(($path.Equals("")) -Or ($path.Contains($azureauthDirectory)) -Or (($azureauthSource -ne $null) -And $azureauthSource.Contains($path)))){
+                if($newPath.Equals("")){
+                    $newPath = -join("${newPath}","${path}")
+                }
+                else{
+                    $newPath = -join("${newPath}",";","${path}")
+                }

```

Comment:
From discussion offline - see if we can copy elements that do not match one of the two dirs to a new array, and then append the version path to that array, and call join at the end to join the array on `;`, instead of string allocating inside the for loop.